src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi

```diff
@@ -80,6 +81,8 @@ cdef class PollerCompletionQueue(BaseCompletionQueue):
         self._loop.remove_reader(self._read_socket)
         # TODO(https://github.com/grpc/grpc/issues/22365) perform graceful shutdown
         grpc_completion_queue_shutdown(self._cq)
+        while not self._shutdown:
+            self._poller_thread.join(timeout=_POLL_AWAKE_INTERVAL_S)
```

Comment:
I'm assuming you added the timeout for signal handling? If that's the case, [I think `cpython` already takes care of this for you.](https://github.com/python/cpython/blob/master/Modules/_threadmodule.c#L71) No need to set a timeout.

Reply:
`acquire_timed` seems mostly called by lock acquiring with a timeout argument (might be wrong), otherwise we don't need the `_common.wait` function in the existing stack. 

he timeout is given the control back to interpreter, so it can 1) perform work between Python frames (like GC, or other state change); 2) tell CPython to use `acquire_timed`.

Reply:
Still, it allows signal to interrupt the call, I think your statement is correct (see https://github.com/python/cpython/blob/master/Modules/_threadmodule.c#L67).